### PR TITLE
Lease controller - GET

### PR DESCRIPTION
**Pull request by AmudaPalani**
## Proposed changes

Use new data model for leases GET controller

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (changes to code, which do not change application behavior)

## Checklist

- [ ] I have filled out this PR template
- [ ] I have read the [CONTRIBUTING](https://github.com/Optum/dce/blob/master/docs/CONTRIBUTING.md) doc
- [ ] I have added automated tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (`README.md`, inline comments, etc.)
- [ ] I have updated the `CHANGELOG.md` under a `## next` release, with a short summary of my changes

### CODE REVIEW

File: cmd/lambda/leases/get_test.go [hunk 1]
```diff
@@ -1,66 +1,163 @@
...
+	}
+	tests := []struct {
+		name     string
+		expResp  response
+		leaseID  string
+		retLease *lease.Lease
+		retErr   error
+	}{
+		{
+			name:    "success",
```
Comment (joshmarsh) [inline cmd/lambda/leases/get_test.go]: Making these more descriptive would make the test easier to parse. It's probably also good to be somewhat consistent with how descriptive the names of the non-table driven tests are below.

File: cmd/lambda/leases/main.go [hunk 2]
```diff
@@ -31,21 +45,33 @@ const (
...
-	maxLeaseBudgetAmount     *float64
-	maxLeasePeriod           *int64
-	defaultLeaseLengthInDays *int
+	// Soon to be deprecated - Legacy support
+	Config             common.DefaultEnvConfig
+	awsSession         *session.Session
+	dao                db.DBer
+	snsSvc             common.Notificationer
+	usageSvc           usage.Service
+	leaseAddedTopicARN string
```
Comment (joshmarsh) [inline cmd/lambda/leases/main.go]: Aren't some of these also in leaseControllerConfiguration?

File: cmd/lambda/leases/main.go [hunk 3]
```diff
@@ -127,6 +147,53 @@ func init() {
...
+		WithLeaseService().
+		Build()
+	if err != nil {
+		panic(err)
+	}
+
+	Services = svcBldr
+
+	//resetQueueURL = Config.GetEnvVar("RESET_SQS_URL", "DefaultResetSQSUrl")
+	leaseAddedTopicARN = Config.GetEnvVar("LEASE_ADDED_TOPIC", "DCEDefaultProvisionTopic")
```
Comment (joshmarsh) [inline cmd/lambda/leases/main.go]: Didn't we already retrieve this [when we unmarshalled the env vars into the Settings struct](https://github.com/Optum/dce/pull/240/files#diff-6433aa3e05979c963e2ea37c8e383bc0R154) above?

File: cmd/lambda/leases/main_test.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+import (
+	"os"
+	"testing"
+
+	"github.com/aws/aws-lambda-go/events"
+)
+
+func TestMain(m *testing.M) {
+	os.Setenv("LEASE_ADDED_TOPIC", "mock-lease-added-topic")
```
Comment (kddejong) [inline cmd/lambda/leases/main_test.go]: Do we need to do this?  Can we just use the defaults?

File: pkg/config/servicebuild.go [hunk 5]
```diff
@@ -291,3 +312,64 @@ func (bldr *ServiceBuilder) createAccountService(config ConfigurationServiceBuil
...
+		return err
+	}
+
+	dataSvcImpl.DynamoDB = dynamodbSvc
+
+	config.WithService(dataSvcImpl)
+	return nil
+}
+
+func (bldr *ServiceBuilder) createLeaseManager(config ConfigurationServiceBuilder) error {
```
Comment (kddejong) [inline pkg/config/servicebuild.go]: Thought we were removing this?

File: pkg/lease/service.go [hunk 6]
```diff
@@ -52,12 +52,14 @@ type Eventer interface {
 
 // Manager manages all the actions against a lease
 type Manager interface {
```
Comment (kddejong) [inline pkg/lease/service.go]: Lease isn't going to have a manager service so this could be deleted.

File: pkg/leasemanager/manager.go [hunk 7]
```diff
@@ -0,0 +1,24 @@
+package leasemanager
```
Comment (kddejong) [inline pkg/leasemanager/manager.go]: ditto.  I don't think this needs to exist.

File: cmd/lambda/leases/main.go (hunk 3 above)
Comment (AmudaPalani) [reply cmd/lambda/leases/main.go]: For now we are supporting both old and new ways of interacting with db layer.

File: cmd/lambda/leases/main.go (hunk 2 above)
Comment (AmudaPalani) [reply cmd/lambda/leases/main.go]: For now we are supporting both old and new ways of interacting with db layer.

File: cmd/lambda/leases/get_test.go (hunk 1 above)
Comment (AmudaPalani) [reply cmd/lambda/leases/get_test.go]: Same as accounts.

File: pkg/leasemanager/manager.go (hunk 7 above)
Comment (AmudaPalani) [reply pkg/leasemanager/manager.go]: fixed

File: pkg/config/servicebuild.go (hunk 5 above)
Comment (AmudaPalani) [reply pkg/config/servicebuild.go]: fixed

File: cmd/lambda/leases/main_test.go (hunk 4 above)
Comment (AmudaPalani) [reply cmd/lambda/leases/main_test.go]: fixed

File: cmd/lambda/leases/get_test.go (hunk 1 above)
Comment (AmudaPalani) [reply cmd/lambda/leases/get_test.go]: Modified test names.

File: pkg/lease/service.go (hunk 6 above)
Comment (AmudaPalani) [reply pkg/lease/service.go]: fixed

File: cmd/lambda/leases/get_test.go [hunk 8]
```diff
@@ -1,66 +1,163 @@
...
+		}
+
+		mockRequest := events.APIGatewayProxyRequest{HTTPMethod: http.MethodGet, Path: "/leases/unique-id"}
 
 		actualResponse, err := Handler(context.TODO(), mockRequest)
 		require.Nil(t, err)
 
 		require.Equal(t, actualResponse.StatusCode, 500, "Returns a 500.")
-		require.Equal(t, actualResponse.Body, "{\"error\":{\"code\":\"ServerError\",\"message\":\"Failed Get on Lease unique-id\"}}", "Returns an error response.")
+		require.Equal(t, MockAPIErrorResponse(http.StatusInternalServerError, "{\"error\":{\"message\":\"unknown error\",\"code\":\"ServerError\"}}\n"), actualResponse, "Returns an error response.")
```
Comment (kddejong) [inline cmd/lambda/leases/get_test.go]: I would prefer what is in the accounts controller here.
```
assert.Equal(t, tt.expResp.StatusCode, resp.StatusCode)
assert.Equal(t, tt.expResp.Body, string(body))
```
or even better would probably be using https://godoc.org/github.com/stretchr/testify/assert#JSONEq

File: cmd/lambda/leases/main.go [hunk 9]
```diff
@@ -127,6 +147,52 @@ func init() {
...
+
+	_, err = svcBldr.
+		// AWS services...
+		WithDynamoDB().
+		WithSTS().
+		WithS3().
+		WithSNS().
+		WithSQS().
+		// DCE services...
+		WithStorageService().
```
Comment (kddejong) [inline cmd/lambda/leases/main.go]: I'm not sure we need this.  I would prefer to add it back as we use it?  But nothing too critical here.

File: pkg/config/config.go [hunk 10]
```diff
@@ -268,3 +269,14 @@ func (config *ConfigurationBuilder) AccountSvc() accountiface.Servicer {
 
 	return accountSvc
 }
+
+// LeaseSvc returns the lease Service for you
+func (config *ConfigurationBuilder) LeaseSvc() leaseiface.Servicer {
```
Comment (kddejong) [inline pkg/config/config.go]: I'm working to rename this to AccountService so we should probably change this to LeaseService.

File: cmd/lambda/leases/get_test.go (hunk 8 above)
Comment (AmudaPalani) [reply cmd/lambda/leases/get_test.go]: fixed

File: cmd/lambda/leases/main.go (hunk 9 above)
Comment (AmudaPalani) [reply cmd/lambda/leases/main.go]: fixed

File: pkg/config/config.go (hunk 10 above)
Comment (AmudaPalani) [reply pkg/config/config.go]: renamed

**Pull request merged**